core/src/components/datetime/utils/parse.ts

```diff
@@ -85,6 +87,7 @@ export function parseDate(val: string | string[] | undefined | null): DatetimePa
 
   if (parse === null) {
     // wasn't able to parse the ISO datetime
+    printIonWarning(`Unable to parse date string: ${val}. Please provide a valid ISO 8601 datetime string.`);
```

Comment:
One thing I am observing with assigning an invalid value, is this message doesn't occur.

e.g.:

Assigning a value of `2021-04-20T14:23:00.000z` (lowercase `z` instead of `Z`)

Results in this error:
> Cannot destructure property 'month' of 'targetValue' as it is undefined.

Reply:
Well, the message does occur, but the case isn't being handled once we get back to datetime.tsx 🤔 

![console](https://i.gyazo.com/117ff9c53cb832e7594abe189675fc52.png)

Reply:
Fixed in https://github.com/ionic-team/ionic-framework/pull/27806/commits/3dd1732a9ce2628dec03ae3c33ee9f68004a4e66 and https://github.com/ionic-team/ionic-framework/pull/27806/commits/607f43cb6695f428b9ba413e790d6da402e63eb7, and then cleaned up in https://github.com/ionic-team/ionic-framework/pull/27806/commits/c57c6febf4f7f4f69db64ce615b95896afeb205e, which also updates `parseDate`'s type signatures to account for this error case.